portal-impl/src/com/liferay/portlet/PortalPreferencesImpl.java

```diff
@@ -310,7 +349,12 @@ protected boolean isCausedByStaleObjectException(Throwable t) {
 		return false;
 	}
 
-	protected void retryableStore(Callable<?> callable) throws Throwable {
+	protected void retryableStore(
+			Callable<?> callable, String namespace, String key)
```

Comment:
retryableStore should simply take already encoded key, then it can just call super.